inventory/byo/hosts.origin.example

```diff
@@ -89,6 +89,21 @@ openshift_release=v1.4
 # Skip upgrading Docker during an OpenShift upgrade, leaves the current Docker version alone.
 # docker_upgrade=False
 
+
+# Upgrade Hooks
+#
+# Hooks are available to run custom tasks at various points during a cluster
+# upgrade. Each hook should point to a file with Ansible tasks defined. Suggest using
+# absolute paths, if not the path will be treated as relative to the file where the
+# hook is actually used.
+#
+# Tasks to run before each master is upgraded:
+# openshift_upgrade_pre_master_hook=/usr/share/custom/pre_master.yml
```

Comment:
Probably need to start defining an interface, so that we can set expectations on what variables/state are available in the hooks.

First questions I have:
- What host(s) are the hook run against
- Is the playbook run as a full playbook include, or a task include?
- What variables are set/supported?
- If there is a hook failure in the middle of a run, how does a user recover?
- Do we expect there to be data returned from the hooks (and possibly passed on to later hooks)?